core/src/lib.rs

```diff
@@ -58,11 +58,19 @@ cfg_client! {
 /// Shared tracing helpers to trace RPC calls.
 pub mod tracing;
 pub use async_trait::async_trait;
-pub use error::Error;
+pub use error::{Error, SubscriptionAcceptRejectError, SubscriptionClosed, SubscriptionEmptyError};
 
 /// JSON-RPC result.
 pub type RpcResult<T> = std::result::Result<T, Error>;
 
+/// The return type of the subscription's method for the rpc server implementation.
+///
+/// **Note**: The error does not contain any data and is discarded on drop.
+pub type SubscriptionResult = Result<(), SubscriptionEmptyError>;
```

Comment:
moved these to `core`, I want to keep initial idea to keep JSON-RPC spec related types in the types crate.